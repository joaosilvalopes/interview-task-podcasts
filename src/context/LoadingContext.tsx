import { createContext } from "react";

type LoadingContextType = [boolean, (loading: boolean) => void];

export default createContext<LoadingContextType>([true, () => {}]);
