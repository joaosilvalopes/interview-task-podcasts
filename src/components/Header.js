import { useContext } from "react";
import styled from "styled-components";

import LoadingContext from "../context/LoadingContext";

import Loader from './Loader';

const Header = styled.header`
    display: flex;
    justify-content: space-between;
    padding: 2.5rem 0 1.6rem;
    margin: 0 3.2rem;
    border-bottom: 0.1rem solid rgba(0, 0, 0, 0.1);
`;

const PageTitle = styled.h1`
    color: blue;
    font-size: 2.8rem;
`;

const _Header = () => {
    const [loading] = useContext(LoadingContext);

    return (
        <Header><PageTitle>Podcaster</PageTitle>{loading  && <Loader /> }</Header>
    );
}

export default _Header;
