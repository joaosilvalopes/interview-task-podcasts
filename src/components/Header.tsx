import { useContext } from "react";
import styled from "styled-components";

import LoadingContext from "../context/LoadingContext";

import Loader from './Loader';

const Header = styled.header`
    display: flex;
    justify-content: space-between;
    padding: 2.5rem 0 1.6rem;
    margin: 0 3.2rem;
    border-bottom: 0.1rem solid rgba(0, 0, 0, 0.1);
`;

const PageTitle = styled.h1`
    color: blue;
    font-size: 2.8rem;
`;

const FadingLoader = styled(Loader).withConfig({ shouldForwardProp: (prop: string | number) => prop !== 'loading' })`
    opacity: ${(props: { loading: true }) => props.loading ? 1 : 0};
    transition: 0.5s opacity ease-out;
`;

const _Header = () => {
    const [loading] = useContext(LoadingContext);

    return (
        <Header><PageTitle>Podcaster</PageTitle><FadingLoader data-testid="loader" loading={loading} /></Header>
    );
}

export default _Header;
