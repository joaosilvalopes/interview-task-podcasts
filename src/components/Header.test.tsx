import { render, screen } from '@testing-library/react';
import LoadingContext from '../context/LoadingContext';

import Header from './Header';

it('should render correctly when loading context passed as true', async () => {
    const { asFragment, getByTestId } = render(<LoadingContext.Provider value={[true, () => {}]}><Header /></LoadingContext.Provider>);
  
    expect(getByTestId('loader')).toBeVisible();
    expect(asFragment()).toMatchSnapshot();
});

it('should render correctly when loading context passed as false', async () => {
    const { asFragment, getByTestId } = render(<LoadingContext.Provider value={[false, () => {}]}><Header /></LoadingContext.Provider>);
  
    expect(getByTestId('loader')).not.toBeVisible();
    expect(asFragment()).toMatchSnapshot();
});
