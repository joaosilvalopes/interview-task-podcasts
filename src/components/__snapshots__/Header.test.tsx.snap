// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly when loading context passed as false 1`] = `
<DocumentFragment>
  <header
    class="sc-hLBbgP bHnltk"
  >
    <h1
      class="sc-eDvSVe ckjSgl"
    >
      Podcaster
    </h1>
    <div
      class="sc-bcXHqe eglgry sc-jSUZER bQyIE"
      data-testid="loader"
    >
      <div
        class="sc-dkrFOg eZaBCl"
      />
      <div
        class="sc-gswNZR eMufKk"
      />
    </div>
  </header>
</DocumentFragment>
`;

exports[`should render correctly when loading context passed as true 1`] = `
<DocumentFragment>
  <header
    class="sc-hLBbgP bHnltk"
  >
    <h1
      class="sc-eDvSVe ckjSgl"
    >
      Podcaster
    </h1>
    <div
      class="sc-bcXHqe eglgry sc-jSUZER lnDLHd"
      data-testid="loader"
    >
      <div
        class="sc-dkrFOg eZaBCl"
      />
      <div
        class="sc-gswNZR eMufKk"
      />
    </div>
  </header>
</DocumentFragment>
`;
